protocol/x/clob/keeper/equity_tier_limit.go

```diff
@@ -117,48 +119,39 @@ func (k Keeper) ValidateSubaccountEquityTierLimitForNewOrder(ctx sdk.Context, or
 		)
 	}
 
-	// Count all the open orders that are on the `MemClob`.
-	equityTierCount := int32(k.MemClob.CountSubaccountOrders(ctx, subaccountId, filter))
-
-	// Include the number of stateful orders that exist outside of the `MemClob`. This includes the number of
-	// untriggered conditional orders and during `DeliverTx` we add the count of to be committed stateful orders
-	// while in `CheckTx` we add the count of uncommitted stateful orders.
+	equityTierCount := uint32(0)
 	if order.IsStatefulOrder() {
-		equityTierCount += int32(k.CountUntriggeredSubaccountOrders(ctx, subaccountId, filter))
-		if lib.IsDeliverTxMode(ctx) {
-			equityTierCount += k.GetToBeCommittedStatefulOrderCount(ctx, order.OrderId)
-		} else {
-			equityTierCount += k.GetUncommittedStatefulOrderCount(ctx, order.OrderId)
-		}
-	}
-
-	if equityTierCount < 0 {
-		if lib.IsDeliverTxMode(ctx) {
-			err = fmt.Errorf(
-				"Expected ValidateSubaccountEquityTierLimitForNewOrder for new order %+v to be >= 0. "+
-					"equityTierCount %d, memClobCount %d, (stateful order only) toBeCommittedCount %d.",
-				order,
-				equityTierCount,
-				k.MemClob.CountSubaccountOrders(ctx, subaccountId, filter),
-				k.GetToBeCommittedStatefulOrderCount(ctx, order.OrderId),
-			)
-		} else {
-			err = fmt.Errorf(
-				"Expected ValidateSubaccountEquityTierLimitForNewOrder for new order %+v to be >= 0. "+
-					"equityTierCount %d, memClobCount %d, (stateful order only) uncommittedCount %d.",
-				order,
-				equityTierCount,
-				k.MemClob.CountSubaccountOrders(ctx, subaccountId, filter),
-				k.GetUncommittedStatefulOrderCount(ctx, order.OrderId),
-			)
+		// For stateful orders we get the stateful order count which represents how many long term and
+		// triggered conditional orders exist in state. We add to that all untriggered conditional orders.
+		// If this is `CheckTx` then we must also add the number of uncommitted stateful orders that this validator
+		// is aware of (orders that are part of the mempool but have yet to proposed in a block).
+		equityTierCount = k.GetStatefulOrderCount(ctx, order.OrderId)
+		equityTierCount += k.CountUntriggeredSubaccountOrders(ctx, subaccountId, filter)
+		if !lib.IsDeliverTxMode(ctx) {
```

Comment:
We don't need to gate this, right? count is just gonna be zero for deliver tx 

probably improves readability

Reply:
I think it makes sense to gate this as we don't want to count this number when in `DeliverTx` and keeps the logic of what is meant to be summed clearer instead of having implicit knowledge or a comment that we expect this to be zero during `DeliverTx`.

Reply:
Yeah personally I think this is more readable if it's gated, since then you don't need to worry about the implementation of `GetUncommittedStatefulOrderCount` when thinking about this function in the context of `DeliverTx`

In the longer term, a larger readability improvement is likely splitting out this function into two, one to use in `DeliverTx` and another to use in `CheckTx`.